src/routes/home/components/ProfessionDetails.tsx

```diff
@@ -0,0 +1,100 @@
+import { CommonActions, useFocusEffect, useNavigation } from '@react-navigation/native'
+import { StackNavigationProp } from '@react-navigation/stack'
+import React, { ReactElement, useCallback, useEffect, useState } from 'react'
+import * as Progress from 'react-native-progress'
+import styled from 'styled-components/native'
+
+import Button from '../../../components/Button'
+import { BUTTONS_THEME, EXERCISES } from '../../../constants/data'
+import { Discipline, Document } from '../../../constants/endpoints'
+import labels from '../../../constants/labels.json'
+import theme from '../../../constants/theme'
+import { loadDocuments } from '../../../hooks/useLoadDocuments'
+import useReadNextExercise from '../../../hooks/useReadNextExercise'
+import useReadProgress from '../../../hooks/useReadProgress'
+import { RoutesParams } from '../../../navigation/NavigationTypes'
+import { childrenLabel } from '../../../services/helpers'
+import { reportError } from '../../../services/sentry'
+import { ButtonContainer, NumberText, UnitText } from './DisciplineCard'
+
+const ProgressContainer = styled.View`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  padding: ${props => props.theme.spacings.sm} 0 ${props => props.theme.spacings.xs};
+`
+
+interface PropsType {
+  discipline: Discipline
+}
+
+const ProfessionDetails = ({ discipline }: PropsType): ReactElement => {
+  const { data: nextExercise, refresh: refreshNextExercise } = useReadNextExercise(discipline)
+  const { data: progress, refresh: refreshProgress } = useReadProgress(discipline)
+
+  const moduleAlreadyStarted = progress !== null && progress !== 0
+  const [documents, setDocuments] = useState<Document[] | null>(null)
+  const navigation = useNavigation<StackNavigationProp<RoutesParams, 'Home'>>()
```

Comment:
I decided to use useNavigation() hook here, as I wanted to avoid passing prop to deeply nested child, as the docs say `useNavigation is a hook which gives access to navigation object. It's useful when you cannot pass the navigation prop into the component directly, or don't want to pass it in case of a deeply nested child.`

If we decide to pass the method as you suggest, we need to add a optional prop `navigateToNextExercise()` to `DisciplineCard` and then pass it to `ProfessionDetails` and then check if the prop exists.

Which way do you prefer?